class LoggedError(ValueError):
    """
    Специальное исключение, оно не логируется декораторами.
    Используется для подмены оригинальных исключений в задекорированных логгером функциях, если настройка "original_exceptions" установлена в положении False.
    Это необходимо, чтобы исключение не логировалось многократно, пока поднимается по стеку вызова функций. Оригинальное исключение будет залогировано только в первый раз, заменено данным, после чего будет игнорироваться.
    Установите настройку "original_exceptions" в положение True, если логика работы вашей программы подразумевает реагирование на конкретные типы исключений.
    Вы также можете поднимать это исключение намеренно, если не хотите, чтобы оно было залогировано. Это может вам пригодиться, если вы, к примеру, не хотите засорять хранилище лишними логами при отладке.
    """
    pass

class IncorrectUseOfTheDecoratorError(ValueError):
    """
    Когда в декоратор передали что-то не то.
    """
    pass

class DoubleSettingError(ValueError):
    """
    Некоторые поля настроек пользователю может быть запрещено изменять дважды.
    Если он это делает, поднимается данное исключение.
    """
    pass

class AfterStartSettingError(ValueError):
    """
    Поднимается при попытке изменить настройку, которую запрещено изменять после записи первого лога.
    """
    pass

class RewritingLogError(RuntimeError):
    """
    Поднимается при попытке одного из обработчиков отредактировать запись лога.
    """
    pass

class HandlerNotFoundError(ValueError):
    """
    Поднимается при попытке получить обработчик по ключу, который ранее не был зарегистрирован.
    """
    pass
